src/script_interface/Context.hpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2020 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef ESPRESSO_CONTEXT_HPP
+#define ESPRESSO_CONTEXT_HPP
+
+#include "ObjectHandle.hpp"
+#include "Variant.hpp"
+
+#include <boost/utility/string_ref.hpp>
+
+#include <memory>
+
+namespace ScriptInterface {
+class Context : public std::enable_shared_from_this<Context> {
```

Comment:
can you explain why ```std::enable_shared_from_this``` is needed here?

Reply:
Do you now what it does?

Reply:
yes

Reply:
Ok, it's needed here to that all the `ObjectHandle`s in one context share ownership of the same context. E.g. if you make a new handle in a context, it should share ownership of that context.

Reply:
Okay maybe you should add a docstring to the class explaining what the role of a ```Context``` is here